atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -93,30 +98,38 @@ void SpellCheckClient::SpellCheckText(
     const base::string16& text,
     bool stop_at_first_result,
     std::vector<blink::WebTextCheckingResult>* results) {
-  if (text.length() == 0 || spell_check_.IsEmpty())
+  if (text.empty() || spell_check_.IsEmpty())
     return;
 
-  base::string16 word;
-  int word_start;
-  int word_length;
   if (!text_iterator_.IsInitialized() &&
       !text_iterator_.Initialize(&character_attributes_, true)) {
       // We failed to initialize text_iterator_, return as spelled correctly.
       VLOG(1) << "Failed to initialize SpellcheckWordIterator";
       return;
   }
 
-  base::string16 in_word(text);
-  text_iterator_.SetText(in_word.c_str(), in_word.size());
+  if (!contraction_iterator_.IsInitialized() &&
+      !contraction_iterator_.Initialize(&character_attributes_, false)) {
+    // We failed to initialize the word iterator, return as spelled correctly.
+    VLOG(1) << "Failed to initialize contraction_iterator_";
+    return;
+  }
+
+  text_iterator_.SetText(text.c_str(), text.size());
```

Comment:
It looks like `text_iterator_` is initialized here and only used here. Could we make it a local variable?

Reply:
I suppose initialization of the iterator is expensive, that's why it's kept as a member and reused on invocations of `SpellCheckText`.

Reply:
I suppose initialization of the iterator is expensive, that's why it's kept as a member and reused on invocations of `SpellCheckText`.